### DISCUSSION TITLE
Flow, trigger Send Notification For Each result

### DISCUSSION THREAD

**Discussion by JaimeBulaGooten** [directus/directus#14676] [Archive]
Hi Team, 

Currently I have a filtered Read Step in Flow, which gives me an array as a result. 

How can I Iterate over each result, and send a Notification per result in the next Step?

The current array is something like 
```

[
{Id:"000XX"},{Id:"000XX1"},{Id:"000XX2"}
]

```
but in the next step, writing `{{$last.id}}` fails. Again, is there a way to send a Notification for each item in the array of a previous  Read Operation.

**Comment by rijkvanzanten** [comment]
Heya! This is actually something that's being discussed/worked on in https://github.com/directus/directus/issues/14273 🙂

**Reply by JaimeBulaGooten** [reply]
@rijkvanzanten Yes it is... it is exactly the same issue.

**Comment by azrikahar** [comment]
This should now be made possible with #14694 and #15101, where the latter allows us to write a little script to change array of objects such as:

```json
[
	{ "user_created": "dcd6b1bd-dac6-4f26-b3fb-a90a14e07fd5" },
	{ "user_created": "27131a59-5d6f-4f2a-bdf6-f915a8df9fb0" }
]
```

into array of IDs such as:

```json
["dcd6b1bd-dac6-4f26-b3fb-a90a14e07fd5", "27131a59-5d6f-4f2a-bdf6-f915a8df9fb0"]
```

for operations that receives an array of IDs to work.


Quick example: https://github.com/directus/directus/pull/15101#issuecomment-1219877623 (Example shows usage of `[...new Set(userIds)]` to make sure they are unique so users doesn't get sent multiple notifications)
